cv2.error: in function 'cv::cornerSubPix'

### System Information

Python version: 3.8.18 
OpenCV python version: 4.9.0
Operating System / Platform: Windows-10-10.0.19045-SP0

### Detailed description


### Detailed description

PLEASE TRY IT AGAIN! I ADD AN INPUT VALIDATION
In this case, when I invoke cv2.cornerSubPix by these inputs, it reported:
cv2.error: OpenCV(4.9.0) D:\a\opencv-python\opencv-python\opencv\modules\imgproc\src\cornersubpix.cpp:99: error: (-215:Assertion failed) Rect(0, 0, src.cols, src.rows).contains(cT) in function 'cv::cornerSubPix'

### Steps to reproduce

```
import cv2
import numpy as np

src = np.zeros((500, 500), dtype=np.uint8)

corners = np.array([
    [63.23869, 233.98373],
    [24.23986, 21.29863],
    [123.34534, 499.52298],
    [225.45345, 56.35624],
    [32.87236, 217.28916],
    [124.59836, 166.02876],
], dtype=np.float32).reshape(-1, 1, 2)

h, w = src.shape
valid_corners = np.array([c for c in corners if (0 <= c[0][0] < w) and (0 <= c[0][1] < h)])

winSize = (3, 3)
zeroZone = (-1, -1)
criteria = (cv2.TERM_CRITERIA_EPS + cv2.TERM_CRITERIA_MAX_ITER, 20, 0.1)

refined_corners = cv2.cornerSubPix(src, valid_corners, winSize, zeroZone, criteria)



### Issue submission checklist

- [X] I report the issue, it's not a question
- [X] I checked the problem with documentation, FAQ, open issues, forum.opencv.org, Stack Overflow, etc and have not found any solution
- [X] I updated to the latest OpenCV version and the issue is still there
- [X] There is reproducer code and related data files (videos, images, onnx, etc)

--- a/modules/imgproc/src/cornersubpix.cpp
+++ b/modules/imgproc/src/cornersubpix.cpp
@@ -96,14 +96,11 @@ void cv::cornerSubPix( InputArray _image, InputOutputArray _corners,
     for( int pt_i = 0; pt_i < count; pt_i++ )
     {
         Point2f cT = corners[pt_i], cI = cT;
-
         if (!Rect(0, 0, src.cols, src.rows).contains(cT))
         {
             CV_Error(Error::StsOutOfRange,
                     "cornerSubPix: initial corner is outside the image.");
         }
- 
-        
         int iter = 0;
         double err = 0;
 

